core/domain/prod_validation_jobs_one_off.py

```diff
@@ -110,67 +351,3183 @@ def validate(cls, item):
+
+        Args:
+            item: ConfigPropertySnapshotMetadataModel to validate.
+        """
+        _, config_property_model_tuples_list = cls.external_models[
+            'config_property_ids']
+
+        version = item.id[item.id.rfind('-') + 1:]
+        for (_, config_property_model) in config_property_model_tuples_list:
+            if int(config_property_model.version) < int(version):
+                cls.errors['config property model version check'].append((
+                    'Entity id %s: ConfigProperty model corresponding to '
+                    'id %s has a version %s which is less than the version %s '
+                    'in snapshot metadata model id' % (
+                        item.id, config_property_model.id,
+                        config_property_model.version, version)))
+
+    @classmethod
+    def _get_custom_validation_functions(cls):
+        return [cls._validate_config_property_model_version_from_item_id]
+
+
+class ConfigPropertySnapshotContentModelValidator(BaseModelValidator):
+    """Class for validating ConfigPropertySnapshotContentModel."""
+
+    @classmethod
+    def _get_model_id_regex(cls, item):
+        return '.'
+
+    @classmethod
+    def _get_json_properties_schema(cls, item):
+        return {}
+
+    @classmethod
+    def _get_model_domain_object_instances(cls, item):
+        return []
+
+    @classmethod
+    def _get_commit_cmd_domain_class(cls):
+        return None
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return {
+            'config_property_ids': (
+                config_models.ConfigPropertyModel,
+                [item.id[:item.id.find('-')]]),
+        }
+
+    @classmethod
+    def _validate_config_property_model_version_from_item_id(cls, item):
+        """Validate that config property model corresponding to snapshot
+        content model has a version greater than or equal to the in item.id.
+
+        Args:
+            item: ConfigPropertySnapshotContentModel to validate.
+        """
+        _, config_property_model_tuples_list = cls.external_models[
+            'config_property_ids']
+
+        version = item.id[item.id.rfind('-') + 1:]
+        for (_, config_property_model) in config_property_model_tuples_list:
+            if int(config_property_model.version) < int(version):
+                cls.errors['config property model version check'].append((
+                    'Entity id %s: ConfigProperty model corresponding to '
+                    'id %s has a version %s which is less than the version %s '
+                    'in snapshot content model id' % (
+                        item.id, config_property_model.id,
+                        config_property_model.version, version)))
+
+    @classmethod
+    def _get_custom_validation_functions(cls):
+        return [cls._validate_config_property_model_version_from_item_id]
+
+
+class SentEmailModelValidator(BaseModelValidator):
```

Comment:
As a general note, you are welcome to split this file and associated tests into core/domain/prod_validation_jobs/prod_validation_jobs_email_one_off.py (or whatever a suitable name would be) and corresponding tests, if you like. (OK to do in the future, too, or not at all.)

Reply:
I like the idea of splitting up the jobs since it will make the code shorter and hence more readable. I will do this in a separate PR once this goes in.